### PR TITLE
Add equity tiers in configs output

### PR DESCRIPTION
**Pull request by aforaleka**
need to render an equity tier table in FE similar to this, with tier 1..6
Screenshot 2024-05-28 at 11 48 58 AM

we're also removing equity tier limit validation on the FE, which will be done in another PR. therefore ignoring any duplicated processing logic here

### CODE REVIEW

File: src/commonMain/kotlin/exchange.dydx.abacus/output/Configs.kt [hunk 1]
```diff
@@ -247,6 +249,105 @@ data class FeeTier(
     }
 }
 
+@JsExport
+@Serializable
+data class EquityTiers(
+    val shortTermOrderEquityTiers: List<EquityTier>,
```
Comment (moo-onthelawn) [inline src/commonMain/kotlin/exchange.dydx.abacus/output/Configs.kt:255]: hmm should this more accurately be `IList<EquityTier>`?

File: src/commonMain/kotlin/exchange.dydx.abacus/output/Configs.kt [hunk 2]
```diff
@@ -247,6 +249,105 @@ data class FeeTier(
...
+                    parser.asMap(item)?.let {
+                        val tier = existing?.getOrNull(i)
+                        val nextTierData = parser.asMap(nextItem)
+                        EquityTier.create(tier, parser, it, i + 1, nextTierData)?.let { equityTier ->
+                            equityTiers.add(equityTier)
+                        }
+                    }
+                }
+                return equityTiers
+            }
```
Comment (moo-onthelawn) [inline src/commonMain/kotlin/exchange.dydx.abacus/output/Configs.kt:299]: nit: should we also add an invalid log here

**Review by moo-onthelawn — COMMENTED** [review]
can we also replace the `Map<String, Any>?` typings in the codebase (i.e.  in `EquityTiersProcessor`) and the existing `EquityTier` type in `TradeInputDataValidator` (otherwise it gets a lil confusing to read)?

File: src/commonMain/kotlin/exchange.dydx.abacus/output/Configs.kt (hunk 1 above)
Comment (aforaleka) [reply src/commonMain/kotlin/exchange.dydx.abacus/output/Configs.kt:255]: mmm it seems like they're aliases so IList = List? but i'll change it!

File: v4_abacus.podspec [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |spec|
     spec.name                     = 'v4_abacus'
-    spec.version                  = '1.7.32'
+    spec.version                  = '1.7.33'
```
[the remark this answers is not in the document]
Comment (aforaleka) [reply v4_abacus.podspec:3]: yeee i'll rebase last minute before merge so i know which version to bump to

File: src/commonMain/kotlin/exchange.dydx.abacus/output/Configs.kt [hunk 4]
```diff
@@ -247,6 +249,106 @@ data class FeeTier(
...
+            return null
+        }
+
+        internal fun create(
+            existing: IList<EquityTier>?,
+            parser: ParserProtocol,
+            data: List<*>?
+        ): IList<EquityTier>? {
+            data?.let {
+                val equityTiers = iMutableListOf<EquityTier>()
```
Comment (yogurtandjam) [inline src/commonMain/kotlin/exchange.dydx.abacus/output/Configs.kt:286]: [nit]: 'creating {object name}' logs

**Review by yogurtandjam — APPROVED** [review]
LGTM

**Pull request merged**